agent/internal/client/methods.go

```diff
@@ -583,6 +584,29 @@ func (c *communicatorImpl) AttachFiles(ctx context.Context, taskData TaskData, t
 	return nil
 }
 
+func (c *communicatorImpl) SetDownstreamParams(ctx context.Context, downstreamParams []patch.Parameter, taskId string) error {
+	if len(downstreamParams) == 0 {
```

Comment:
Maybe this check would be more appropriate in the command, to only call `comm.SetDownstreamParams` when there are params to send, as opposed to the communicator being responsible for it.
🤷‍♂️